api/src/resources/comment.rs

```diff
@@ -501,13 +501,33 @@ pub struct EntitySpan {
 }
 
 #[derive(Debug, PartialEq, Clone, Deserialize, Serialize)]
-pub struct EntityKind(pub String);
+pub struct EntityName(pub String);
 
-impl FromStr for EntityKind {
+#[derive(Debug, Serialize, Deserialize)]
+pub struct EntityDef {
+    name: EntityName,
+    title: String,
+    inherits_from: Vec<String>,
+    trainable: bool,
+}
+
+#[derive(Debug, Default, Serialize, Deserialize)]
+pub struct EntityDefs(Vec<EntityDef>);
+
+impl EntityDefs {
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+}
+
+impl FromStr for EntityDefs {
     type Err = Error;
 
     fn from_str(string: &str) -> Result<Self> {
-        Ok(EntityKind(string.to_owned()))
+        serde_json::from_str(string).map_err(|error: JsonError| Error::BadEntityDef {
+            entity_def: string.to_owned(),
+            source: error,
+        })
     }
 }
```

Comment:
`EntityDef` and `EntityDefs` seem like they're probably more a concern for `dataset.rs` than `comment.rs`?

Reply:
Good point, I moved it :+1: